### ISSUE TITLE
Request: auto-spacer

### ISSUE THREAD

**Issue by fire-eggs** [osen/FL_Flex#2]
The need to add FL_Box "spacers" between items (e.g. FL_Flex::ROWs) is kinda awkward. Especially when populating a Fl_Flex::COLUMN. Having an auto-spacer setting (e.g. `setSpacer(int val)`) might be useful. For a column it would add the requested vertical space between children; for a row it would add the requested horizontal space.

I'm liking this class! I nicely re-created a Tkinter GUI of mine pretty quickly:

![image](https://user-images.githubusercontent.com/9809727/95360122-f7c70a80-0898-11eb-8b44-11e7d91c54ee.png)

Note the nice spacing between the Home/Prev/etc buttons. I achieved that by exposing Fl_Flex::margin and setting a value. Works well for a ROW but not so well for a COLUMN.

**Comment by osen** [comment]
Yes, agreed, this will be useful. As you found there is some initial work with Fl_Flex::margin.

Unfortunately I didn't get round to finishing it. I was delaying it until I implemented a "minimum size" kind of system (i.e if you resize the window as small as possible, components will not allow themselves to shrink to 0 (or negative in size). Unfortunately my last attempt at this failed so I will try that again ;)

::margin should have worked for both, I will have a look at why it seems to be having issues for COLUMN.

Thanks!
